week3/yourChoice/src/components/Content/Random.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import styled, { keyframes } from 'styled-components';
+
+import ContentWrapper from '../Layout/ContentWrapper';
+import { SONG_TITLE } from '../../assets/song_data/songData';
+import Result from '../Layout/Result';
+import { StepBtncontainer, ToStartButton } from './style/ContentStyles';
+
+// 랜덤 content
+const Random = (props) => {
+  const [timer, setTimer] = useState(3);
+  const [randomImgSrc, setRandomImgSrc] = useState('');
+  const [randomSongTitle, setRandomSongTitle] = useState('');
+
+  useEffect(() => {
+    const count = setInterval(() => {
+      setTimer((timer) => timer - 1);
+    }, 1000);
+
+    if (timer === 0) {
+      clearInterval(count);
+    }
+
+    return () => {
+      clearInterval(count);
+    };
+  }, [timer]);
+
+  useEffect(() => {
+    if (timer === 0) {
+      randomSong();
+    }
+  }, [timer]);
+
+  function randomSong() {
+    let result = [];
+    for (let i = 0; i < 3; i++) {
+      const randomNum = Math.floor(Math.random() * 3);
+      result.push(randomNum);
+    }
+    let randomImgSrc = `src/assets/img/p${result[0]}_${result[1]}_${result[2]}.jpeg`;
+    let randomSongTitle = SONG_TITLE[result[0]][result[1]][result[2]];
+
+    setRandomImgSrc(randomImgSrc);
+    setRandomSongTitle(randomSongTitle);
```

Comment:
사소한 부분이지만! state를 필요한 만큼 많이 만드는 습관은 추후 조금 더 큰 프로젝트를 구현할 때 지끈지끈...두통을 유발할 수 있기 때문에!! _(상태관리가 존재하는 이유!!)_ 최대한 **구조화**하여 만드는 습관 들여놓으면 좋습니다!! 
예를들어 이 두 state도, 결국 `결과 아이템`이라는 **하나의 주제**를 보여주기 위한 세부정보들이기 때문에 하나의 객체로 묶어주는게 직관성, 효율성 측면에서 조금 더 좋겠죠?!? 

```suggestion
    setRandomItem({
        imgSrc : `src/assets/img/p${result[0]}_${result[1]}_${result[2]}.jpeg`,
        title : SONG_TITLE[result[0]][result[1]][result[2]]
      });
```